### PR TITLE
Add piper backend

### PR DESCRIPTION
**Pull request by PeterD1524**
## Explanation

I am working on [LFX Workspace: Support piper as a new backend of the WASI-NN WasmEdge plugin #3483](https://www.github.com/WasmEdge/WasmEdge/issues/3483).
This PR adds a tutorial to build the piper plugin.

## Related issue

https://github.com/WasmEdge/WasmEdge/issues/3483

## What type of PR is this

> /kind documentation

### CODE REVIEW

**Comment by alabulei1** [comment]
Hello, I am a [code review bot](https://github.com/flows-network/github-pr-summary/) on [flows.network](https://flows.network/).

It could take a few minutes for me to analyze this PR. Relax, grab a cup of coffee and check back later. Thanks!

File: docs/contribute/source/plugin/wasi_nn.md [hunk 1]
```diff
@@ -277,22 +277,29 @@ cmake --build build
 cmake --install build
 ```
 
-#### Apple Silicon Model
+### Appendix
```
Comment (mhmohona) [inline docs/contribute/source/plugin/wasi_nn.md:280]: The main difference between Intel and Apple Silicon builds is the `-DWASMEDGE_PLUGIN_WASI_NN_GGML_LLAMA_METAL` flag, which is set to `OFF` for Intel and `ON` for Apple Silicon. Thus we cannot say `Apple Silicon Model` part is duplicate.

**Review by mhmohona — CHANGES_REQUESTED** [review]
Thank you for updating doc! ✨

File: docs/contribute/source/plugin/wasi_nn.md (hunk 1 above)
Comment (PeterD1524) [reply docs/contribute/source/plugin/wasi_nn.md:280]: The original has two identical `Apple Silicon Model` parts. One is in the MacOS section after `Intel Model` part and the other is in the Linux section after `General Linux without any acceleration framework` part.
I removed the one in the Linux section after `General Linux without any acceleration framework` part.
The diff is a little confusing.

Comment (PeterD1524) [reply docs/contribute/source/plugin/wasi_nn.md:280]: Screenshots from website:

The first one:

image

The second one:

image

Comment (PeterD1524) [reply docs/contribute/source/plugin/wasi_nn.md:280]: Maybe I should open another PR for this? because piper backend has not merged into wasmedge

Comment (hydai) [reply docs/contribute/source/plugin/wasi_nn.md:280]: @PeterD1524 
Let's separate this into another PR. We would like to merge the Piper backend after merging the main PR.

Comment (PeterD1524) [reply docs/contribute/source/plugin/wasi_nn.md]: PR for fixing docs: https://github.com/WasmEdge/docs/pull/239

**Comment by hydai** [comment]
Hi @PeterD1524 
Since you updated the same file in another PR, you will need to update this to fix the conflicts. Thanks.

**Review by hydai — APPROVED** [review]

**Pull request merged**